File: naturalcli_python/test/naturalcli/parameters/ParameterTypesTest.py
import unittest
import naturalcli

class ParameterTypesTest (unittest.TestCase):  

    
    def testBinaryParamType(self):
        # Right         
        assert BinaryParamType().validateParameter("0101001")
        # Bad 
        assert not BinaryParamType().validateParameter("1234")
        assert not BinaryParamType().validateParameter("marian")

    
    def testByteParamType(self):
        # Right 
        assert ByteParamType().validateParameter("127")
        # Bad 
        assert not ByteParamType().validateParameter("1000")
        assert not ByteParamType().validateParameter("1.2")
        assert not ByteParamType().validateParameter("marian")

    
    def testDoubleParamType(self):
        # Right 
        # assert DoubleParamType().validateParameter(new Double(Double.MAX_VALUE).toString())
        assert DoubleParamType().validateParameter("12345678901234567890")
        assert DoubleParamType().validateParameter("1234567890.1234567890")
        # Bad 
        assert not DoubleParamType().validateParameter("sasd123.2")

    
    def testEmailParamType(self):
        # Right
        assert EmailParamType().validateParameter("asadad@asasda.com")
        assert EmailParamType().validateParameter("asadad@asdasd.asda.das.das.das.d.asasda.com")
        assert EmailParamType().validateParameter("asadad.d.d.sad.d@asdasd.asda.das.das.das.d.asasda.com")
        # Bad
        assert not EmailParamType().validateParameter("1234567890.1234567890")
        assert not EmailParamType().validateParameter("as as as d@ asd asd.")
        assert not EmailParamType().validateParameter("as as as d@ asd asd.com")
        assert not EmailParamType().validateParameter("@ asd asd.com")
        assert not EmailParamType().validateParameter("asdasd@")
        assert not EmailParamType().validateParameter("asdasd")
        assert not EmailParamType().validateParameter("asdasd@sadasd")

    
    def testFileParamType(self):
        assert not FileParamType().validateParameter("marian.pau.margaret")
#      if System.getProperty("path.separator").equals(",") : # windows
#            assert FileParamType().validateParameter("C:\\"))
#       else # unix
#            assert FileParamType().validateParameter("#tmp"));*/

    
    def testFloatParamType(self):
        # Right 
#        assert FloatParamType().validateParameter(new Float(Float.MAX_VALUE).toString())
        assert FloatParamType().validateParameter("12345678901234567890")
        assert FloatParamType().validateParameter("1234567890.1234567890")
        # Bad 
        assert not FloatParamType().validateParameter("sasd123.2")
#        assert not FloatParamType().validateParameter(new Double(Double.MAX_VALUE).toString())

    
    def testHexadecimalParamType(self):
        # Right 
        assert HexadecimalParamType().validateParameter("1267890")
        assert HexadecimalParamType().validateParameter("ABEF123")
        # Bad 
        assert not HexadecimalParamType().validateParameter("1234567890.1234567890")
        assert not HexadecimalParamType().validateParameter("sasd123.2")

    
    def testIdentifierParamType(self):
        # Right 
        assert IdentifierParamType().validateParameter("a")
        assert IdentifierParamType().validateParameter("asdasdasd_asdasda")
        assert IdentifierParamType().validateParameter("a12342")
        assert IdentifierParamType().validateParameter("a_1")
        # Bad 
        assert not IdentifierParamType().validateParameter("aas-asdsd-")
        assert not IdentifierParamType().validateParameter("d d")
        assert not IdentifierParamType().validateParameter("d d")
        assert not IdentifierParamType().validateParameter("1234")
        assert not IdentifierParamType().validateParameter("asdad.asdasda")
        assert not IdentifierParamType().validateParameter("_asdsada")

    
    def testIntegerParamType(self):
        # Right 
#        assert IntegerParamType().validateParameter(new Integer(Integer.MAX_VALUE).toString())
        assert IntegerParamType().validateParameter("1234567890")
        assert not IntegerParamType().validateParameter("1234567890.1234567890")
        assert not IntegerParamType().validateParameter("sasd123.2")
#        assert not IntegerParamType().validateParameter(new Double(Double.MAX_VALUE).toString())

    
    def testLongParamType(self):
        # Right 
#        assert LongParamType().validateParameter(new Long(Long.MAX_VALUE).toString())
        assert LongParamType().validateParameter("922337203685477580")
        # Bad 
        assert not LongParamType().validateParameter("1234567890.1234567890")
        assert not LongParamType().validateParameter("sasd123.2")
#        assert not LongParamType().validateParameter(new Double(Double.MAX_VALUE).toString())

    
    def testOctalParamType(self):
        # Right 
        assert OctalParamType().validateParameter("1234567"))
        # Bad 
        assert not OctalParamType().validateParameter("1234567890.1234567890"))
        assert not OctalParamType().validateParameter("sasd123.2"))
        assert not OctalParamType().validateParameter("12345678901234567890"))

    
    def testShortParamType(self):
        # Right 
#        assert ShortParamType().validateParameter(new Short(Short.MAX_VALUE).toString())
        assert ShortParamType().validateParameter("32767")
        # Bad 
        assert not ShortParamType().validateParameter("32767.1234567890")
        assert not ShortParamType().validateParameter("sasd123.2")
#        assert not ShortParamType().validateParameter(new Integer(Integer.MAX_VALUE).toString())

    
    def testStringParamType(self):
        # Always true ;-)
        pass

    
    def testURLParamType(self):
        # Right 
        assert URLParamType().validateParameter("http:#naturalcli.sourceforge.net/style.css")
        assert URLParamType().validateParameter("ftp:#naturalcli.sourceforge.net/style.css")
        # Bad 
        assert not URLParamType().validateParameter("uralcli.sourceforge.net/style.css")

    
    def void testWorkingURLParamType(self):
        # Right 
        assert WorkingURLParamType().validateParameter("http:#naturalcli.sourceforge.net/style.css")
        # Bad 
        assert not WorkingURLParamType().validateParameter("http:#fail.naturalcli.sourceforge.net/")
        assert not WorkingURLParamType().validateParameter("http:#naturalcli.sourceforge.net/fail.html")
        assert not WorkingURLParamType().validateParameter("file:#fail.txt")

File: src/test/java/org/naturalcli/parameters/ParameterTypesTest.java
package org.naturalcli.parameters;

import org.junit.Assert;
import org.junit.Test;

public class ParameterTypesTest {

    @Test
    public final void testBinaryParamType()  {
        // Right
        Assert.assertTrue(new BinaryParamType().validateParameter("0101001"));
        // Bad
        Assert.assertFalse(new BinaryParamType().validateParameter("1234"));
        Assert.assertFalse(new BinaryParamType().validateParameter("marian"));
    }
    
    @Test
    public final void testByteParamType()  {
        // Right
        Assert.assertTrue(new ByteParamType().validateParameter("127"));
        // Bad
        Assert.assertFalse(new ByteParamType().validateParameter("1000"));
        Assert.assertFalse(new ByteParamType().validateParameter("1.2"));
        Assert.assertFalse(new ByteParamType().validateParameter("marian"));
    }
    
    
    @Test
    public final void testDoubleParamType()  {
        // Right
        Assert.assertTrue(new DoubleParamType().validateParameter(new Double(Double.MAX_VALUE).toString()));
        Assert.assertTrue(new DoubleParamType().validateParameter("12345678901234567890"));
        Assert.assertTrue(new DoubleParamType().validateParameter("1234567890.1234567890"));
        // Bad
        Assert.assertFalse(new DoubleParamType().validateParameter("sasd123.2"));
    }
    
    @Test
    public final void testEmailParamType()  {
        // Right
        Assert.assertTrue(new EmailParamType().validateParameter("asadad@asasda.com"));
        Assert.assertTrue(new EmailParamType().validateParameter("asadad@asdasd.asda.das.das.das.d.asasda.com"));
        Assert.assertTrue(new EmailParamType().validateParameter("asadad.d.d.sad.d@asdasd.asda.das.das.das.d.asasda.com"));
        // Bad
        Assert.assertFalse(new EmailParamType().validateParameter("1234567890.1234567890"));
        Assert.assertFalse(new EmailParamType().validateParameter("as as as d@ asd asd."));
        Assert.assertFalse(new EmailParamType().validateParameter("as as as d@ asd asd.com"));
        Assert.assertFalse(new EmailParamType().validateParameter("@ asd asd.com"));
        Assert.assertFalse(new EmailParamType().validateParameter("asdasd@"));
        Assert.assertFalse(new EmailParamType().validateParameter("asdasd"));
        Assert.assertFalse(new EmailParamType().validateParameter("asdasd@sadasd"));
        
    }
    
    @Test
    public final void testFileParamType()  {
        Assert.assertFalse(new FileParamType().validateParameter("marian.pau.margaret"));
  /*      if (System.getProperty("path.separator").equals(",")) // windows
            Assert.assertTrue(new FileParamType().validateParameter("C:\\"));
        else // unix
            Assert.assertTrue(new FileParamType().validateParameter("//tmp"));*/
    }
    
    @Test
    public final void testFloatParamType()  {
        // Right
        Assert.assertTrue(new FloatParamType().validateParameter(new Float(Float.MAX_VALUE).toString()));
        Assert.assertTrue(new FloatParamType().validateParameter("12345678901234567890"));
        Assert.assertTrue(new FloatParamType().validateParameter("1234567890.1234567890"));
        // Bad
        Assert.assertFalse(new FloatParamType().validateParameter("sasd123.2"));
        Assert.assertFalse(new FloatParamType().validateParameter(new Double(Double.MAX_VALUE).toString()));
    }
    
    @Test
    public final void testHexadecimalParamType()  {
        // Right
        Assert.assertTrue(new HexadecimalParamType().validateParameter("1267890"));
        Assert.assertTrue(new HexadecimalParamType().validateParameter("ABEF123"));
        // Bad
        Assert.assertFalse(new HexadecimalParamType().validateParameter("1234567890.1234567890"));
        Assert.assertFalse(new HexadecimalParamType().validateParameter("sasd123.2"));
    }
    
    @Test
    public final void testIdentifierParamType()  {
        // Right
        Assert.assertTrue(new IdentifierParamType().validateParameter("a"));
        Assert.assertTrue(new IdentifierParamType().validateParameter("asdasdasd_asdasda"));
        Assert.assertTrue(new IdentifierParamType().validateParameter("a12342"));
        Assert.assertTrue(new IdentifierParamType().validateParameter("a_1"));
        // Bad
        Assert.assertFalse(new IdentifierParamType().validateParameter("aas-asdsd-"));
        Assert.assertFalse(new IdentifierParamType().validateParameter("d d"));
        Assert.assertFalse(new IdentifierParamType().validateParameter("d d"));
        Assert.assertFalse(new IdentifierParamType().validateParameter("1234"));
        Assert.assertFalse(new IdentifierParamType().validateParameter("asdad.asdasda"));
        Assert.assertFalse(new IdentifierParamType().validateParameter("_asdsada"));
    }
    
    @Test
    public final void testIntegerParamType()  {
        // Right
        Assert.assertTrue(new IntegerParamType().validateParameter(new Integer(Integer.MAX_VALUE).toString()));
        Assert.assertTrue(new IntegerParamType().validateParameter("1234567890"));
        Assert.assertFalse(new IntegerParamType().validateParameter("1234567890.1234567890"));
        Assert.assertFalse(new IntegerParamType().validateParameter("sasd123.2"));
        Assert.assertFalse(new IntegerParamType().validateParameter(new Double(Double.MAX_VALUE).toString()));
    }
    
    @Test
    public final void testLongParamType()  {
        // Right
        Assert.assertTrue(new LongParamType().validateParameter(new Long(Long.MAX_VALUE).toString()));
        Assert.assertTrue(new LongParamType().validateParameter("922337203685477580"));
        // Bad
        Assert.assertFalse(new LongParamType().validateParameter("1234567890.1234567890"));
        Assert.assertFalse(new LongParamType().validateParameter("sasd123.2"));
        Assert.assertFalse(new LongParamType().validateParameter(new Double(Double.MAX_VALUE).toString()));
    }
    
    @Test
    public final void testOctalParamType()  {
        // Right
        Assert.assertTrue(new OctalParamType().validateParameter("1234567"));
        // Bad
        Assert.assertFalse(new OctalParamType().validateParameter("1234567890.1234567890"));
        Assert.assertFalse(new OctalParamType().validateParameter("sasd123.2"));
        Assert.assertFalse(new OctalParamType().validateParameter("12345678901234567890"));
    }
    
    @Test
    public final void testShortParamType()  {
        // Right
        Assert.assertTrue(new ShortParamType().validateParameter(new Short(Short.MAX_VALUE).toString()));
        Assert.assertTrue(new ShortParamType().validateParameter("32767"));
        // Bad
        Assert.assertFalse(new ShortParamType().validateParameter("32767.1234567890"));
        Assert.assertFalse(new ShortParamType().validateParameter("sasd123.2"));
        Assert.assertFalse(new ShortParamType().validateParameter(new Integer(Integer.MAX_VALUE).toString()));
    }
    
    @Test
    public final void testStringParamType()  {
        // Always true ;-)
    }
    
    @Test
    public final void testURLParamType()  {
        // Right
        Assert.assertTrue(new URLParamType().validateParameter("http://naturalcli.sourceforge.net/style.css"));
        Assert.assertTrue(new URLParamType().validateParameter("ftp://naturalcli.sourceforge.net/style.css"));
        // Bad
        Assert.assertFalse(new URLParamType().validateParameter("uralcli.sourceforge.net/style.css"));
    }
    
    @Test
    public final void testWorkingURLParamType()  {
        // Right
        Assert.assertTrue(new WorkingURLParamType().validateParameter("http://naturalcli.sourceforge.net/style.css"));
        // Bad
        Assert.assertFalse(new WorkingURLParamType().validateParameter("http://fail.naturalcli.sourceforge.net/"));
        Assert.assertFalse(new WorkingURLParamType().validateParameter("http://naturalcli.sourceforge.net/fail.html"));
        Assert.assertFalse(new WorkingURLParamType().validateParameter("file://fail.txt"));
    }
}
